code/src/main/kotlin/lexer/Lexer.kt

```diff
@@ -6,34 +6,47 @@ import kotlin.coroutines.experimental.buildSequence
  * The default implementation of the ILexer interface
  */
 class Lexer(private val inputContent: String) : ILexer {
-    private val endLines = listOf("\n","\r\n",";")
-    private val specialChars = listOf("=", "[", "]", "(", ")", "{", "}", ",", "\\", "->", "\\") + endLines
+    // we don't need to add \r\n as every line ending is removed and then each line is appended with \n
+    private val endLines = listOf('\n',';')
+    private val specialChars = listOf('=', '[', ']', '(', ')', '{', '}', ',', '\\', ':') + endLines
     private val endOfLineRegex = "\\r\\n|\\n|\\r".toRegex()
 
     override fun getTokenSequence(): Sequence<PositionalToken> = buildSequence {
         val currentString = StringBuilder()
-        inputContent.split(endOfLineRegex).forEachIndexed { lineNumber, line ->
-            (line + "\n").forEachIndexed { indexNumber, char ->
+        inputContent.split(endOfLineRegex).forEachIndexed lineIterator@{ lineNumber, line ->
+            (line + '\n').forEachIndexed { indexNumber, char ->
                 if (char !in listOf(' ', '\t')) currentString.append(char)
+                if (char == '#'){
+                    yield(PositionalToken(Token.SpecialChar.EndOfLine(), lineNumber, indexNumber))
+                    currentString.setLength(0)
+                    return@lineIterator
+                }
+
                 with(currentString.toString()) {
                     when {
```

Comment:
Consider if an if expression is better for when with 2 cases. Not saying it is, just consider if it is.